gaps/gap_node_descriptor/gap_node_descriptor.md

```diff
@@ -82,23 +82,28 @@ This part explains the steps to be taken when verifying a node descriptor during
 3. Verify that the node descriptor's permission property does not include any permissions that is not granted to the signing certificate. (Details about permission verification can be found in the [Golem certificate structure document](https://github.com/golemfactory/golem-architecture/blob/master/gaps/gap-25_golem_certificates/gap-25_golem_certificates.md#verifying-a-certificate-chain))
 4. Verify that the Computation manifest does not request any features or access that the node is not granted via the node descriptor.
 
+## Rationale
 
-//// stopped here
+### Why do we need node descriptors?
 
-## Rationale
-The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work.
+Node descriptors are one way for providers to trust requestors that their requests are valid and does not intent to cause harm. This is the first KYC solution to allow finer control over outbound and similar sensitive upcoming features.
 
 ## Backwards Compatibility
-All GAPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The GAP **must** explain how the author proposes to deal with these incompatibilities.
+
+When a requestor does not know about node descriptors or the node does not have one, the property in the demand will not be set. It is up to the provider if it will server such requests or not.
+An older provider will not recognize the data in this property and will simply ignore it, reducing its capability to verify the requestor.
 
 ## Test Cases
-Test cases are very useful in summarizing the scope and specifics of a GAP.  If the test suite is too large to reasonably be included inline, then consider adding it as one or more files in `./gaps/gap-draft_title/` directory.
 
-## [Optional] Reference Implementation
-An optional section that contains a reference/example implementation that people can use to assist in understanding or implementing this specification.  If the implementation is too large to reasonably be included inline, then consider adding it as one or more files in `./gaps/gap-draft_title/`.
+The reference implementation contains appropriate tests.
+
+## Reference Implementation
+
+[Golem certificate](https://github.com/golemfactory/golem-certificate) library is available to create, sign, verify node descriptors. The reference implementation of Golem network node [yagna](https://github.com/golemfactory/yagna) will also contain the feature from v0.13.0.
```

Comment:
I wouldn't like to make release promises in the GAP

Reply:
We have it implemented already. So what is the risk? Is it better to have the reference implementation part empty like in other GAPs that are actually implemented?

Reply:
You did not respond in the past days, changed wording.